Feature/start ghg zoom

Bluesmile82 opened the pull request:
This PR adds the data zoom selector to the GHG emissions chart
Missing: We have to position the selector between the chart and the legend. That will need some work on the climate-watch-components.

The change only occurs after the handle is released. The rendering of the chart was not fluid enough.

![image](https://user-images.githubusercontent.com/9701591/78268012-ed057480-7507-11ea-9d82-3f835d8f26f2.png)

kfilip94 commented:
- I think the line on the timeline chart looks slightly different on the design (thicker line, light grey), here it's thinner and blue.
**DESIGN:**
![image](https://user-images.githubusercontent.com/15097138/78361020-98b5cf80-75af-11ea-865b-24384644712c.png)
**IMPLEMENTATION**
![image](https://user-images.githubusercontent.com/15097138/78361112-be42d900-75af-11ea-8352-4a18cd43048b.png)
- I think we should add the tooltip to the handlers while changing the date, we can't see what dates we're choosing

kfilip94 commented on app/javascript/app/components/ghg-emissions/data-zoom/data-zoom-component.js:
```diff
@@ -0,0 +1,122 @@
+  const PADDING = 20;
+  const [position, setPosition] = useState({ min: 0, max: width - PADDING });
+
+  useEffect(() => {
+    if (dataZoomRef.current) {
+      const debouncedSetWidth = debounce(
+        () => setWidth(dataZoomRef.current.offsetWidth),
+        500
+      );
+
+      const refWidth = dataZoomRef.current.offsetWidth;
+      setWidth(refWidth);
+      setPosition({
+        ...position,
+        max: refWidth - PADDING
+      });
+
+      window.addEventListener('resize', debouncedSetWidth);
+      return () => {
+        window.removeEventListener('resize', debouncedSetWidth);
+      };
+    }
+    return undefined;
+  }, [dataZoomRef]);
+
+  const getYear = handleType => {
+    const handleStep = (steps * position[handleType]) / (width - PADDING);
+    const stepData = data && data[Math.floor(handleStep)];
+    return stepData && stepData.x;
+  };
+
+  const handleStop = () => {
+    onYearChange(getYear('min'), getYear('max'));
+  };
+
+  const handleDrag = (ui, handleType) =>
+    setPosition({
+      ...position,
+      [handleType]: position[handleType] + ui.deltaX
+    });
```
Could you move all these functions and hooks to the container component? And leave only render methods here? Also please change the extension of this component file from `.jsx` to `.js`

Bluesmile82 commented:
I will change the line. About the tooltip, what do you think @faustoperez ? I'm not sure if its needed

kfilip94 commented:
Everything else looks great, good job @Bluesmile82, this new slider works very smoothly! :1st_place_medal:
About the slider tooltip, I think we need to give feedback to the user about the date that the slider is currently pointing. Without it, the user can only guess where is the desired date on the timeline. It doesn't have to be a tooltip, let's wait for Fausto's comment here :)

Bluesmile82 commented:
Ready again I included it in the right position, fixed the styles and moved the logic code to the container

![image](https://user-images.githubusercontent.com/9701591/78502256-8929b880-7760-11ea-84cc-913790213a4c.png)

kfilip94 commented:
Hey @Bluesmile82, thank you for your changes!
One last thing that I'd like to ask you: since you've made some changes in `ghg-emissions-component.jsx`, could you move all hooks, handle methods and maybe data objects to the container component `ghg-emissions.js`? We have handle functions there so it's a bit noisy to have them in both files.

Bluesmile82 commented:
Hey! I moved some logic to the container. I prefer to leave the data objects on the component as they will be easier to find. Take a look!
